### PR TITLE
Fix parsing for LOBSTER runs on top of Quantum Espresso

### PR DESCRIPTION
**Pull request by ondracka**

### CODE REVIEW

**Comment by ondracka** [comment]
The test fails now because it needs https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/merge_requests/1128, but it passes properly here locally.

**Comment by ondracka** [comment]
@ladinesa Any chance for a review? :wink:

File: workflowparsers/lobster/parser.py [hunk 1]
```diff
@@ -354,7 +354,7 @@ def translate_lm(lm):
     Quantity('x_lobster_code',
              r'detecting used PAW program... (.*)', repeats=False),
```
Comment (ladinesa) [inline workflowparsers/lobster/parser.py:355]: put here flatten=False

File: workflowparsers/lobster/parser.py [hunk 2]
```diff
@@ -393,6 +393,8 @@ def parse(self, mainfile: str, archive: EntryArchive, logger=None):
                                           '%Y-%m-%d at %H:%M:%S') - datetime.datetime(1970, 1, 1)
         run.time_run = TimeRun(wall_start=date.total_seconds())
         code = mainfile_parser.get('x_lobster_code')
+        if type(code) is list:
```
Comment (ladinesa) [inline workflowparsers/lobster/parser.py:396]: you do not need this is x_lobster_code quantity has flatten=False

File: workflowparsers/lobster/parser.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 import datetime
 import numpy as np
 import ase.io
-from os import path
+from os import path, listdir
```
Comment (ladinesa) [inline workflowparsers/lobster/parser.py:22]: maybe just import os

File: workflowparsers/lobster/parser.py [hunk 4]
```diff
@@ -401,8 +403,16 @@ def parse(self, mainfile: str, archive: EntryArchive, logger=None):
                     structure = ase.io.read(mainfile_path + '/CONTCAR', format="vasp")
                 except FileNotFoundError:
                     logger.warning('Unable to parse structure info, no CONTCAR detected')
+            if code == 'Quantum Espresso':
+                for file in listdir(mainfile_path):
+                    # lobster requires the QE input to have *.scf.in suffix
+                    if file.endswith(".scf.in"):
+                        structure = ase.io.read(mainfile_path + '/' + file, format="espresso-in")
```
Comment (ladinesa) [inline workflowparsers/lobster/parser.py:410]: use os.path.join()

File: workflowparsers/lobster/parser.py [hunk 5]
```diff
@@ -401,8 +403,16 @@ def parse(self, mainfile: str, archive: EntryArchive, logger=None):
                     structure = ase.io.read(mainfile_path + '/CONTCAR', format="vasp")
```
Comment (ladinesa) [inline workflowparsers/lobster/parser.py:403]: also here

**Comment by ladinesa** [comment]
just minor styling then you can merge.

File: workflowparsers/lobster/parser.py (hunk 4 above)
Comment (ladinesa) [reply workflowparsers/lobster/parser.py:410]: we should be able to link this to the qe scf calculation. you can use
```
archive_qe = archive.m_context.resolve_archive(f'../upload/archive/mainfile/{path}')
archive.system = archive_qe.run[0].system[0]
```

Comment (ladinesa) [reply workflowparsers/lobster/parser.py]: great. please merge.

**Comment by ondracka** [comment]
I don't think I have the rights to merge...

**Comment by ondracka** [comment]
Thanks.

**Pull request merged**